### PR TITLE
feature: support training inputs from EFS and FSx

### PR DESCRIPTION
**Pull request by laurenyu**
*Issue #, if available:*

*Description of changes:*
Amazon SageMaker now supports Amazon EFS and Amazon FSx for Lustre file systems as data sources for training machine learning models. This is work from @caxiaohu to support that here.

NB: the resources don't necessarily get cleaned up properly if things fail during setup, but we can track that as a separate issue in our backlog

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://mdk06ma93j.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=b9c7cdaa-e709-4643-bca8-f9a7dc20e226%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://rp2is7byfb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=703e78a4-022c-4409-ad27-6cdab11b8630%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://hu1pg8vleb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=41147245-7025-4737-8397-c3d714a35806%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=9e8908f4-679c-435f-9bca-376f9890ac86%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://mdk06ma93j.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=b4a547c3-31a2-4dad-9b20-8e313227adcc%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://rp2is7byfb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=cd79d9bc-e79a-4a42-860b-49b6e4449edd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://hu1pg8vleb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=4ea67fac-868e-4ff1-8e36-7a92e24edc95%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=c6817c14-cbf9-4627-aaad-2b91a537a3cb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://mdk06ma93j.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=3f9a0c89-f432-4df6-a7c3-2571edf8ae66%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://rp2is7byfb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=b1041c77-691f-42a8-97a6-d8170de82384%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://hu1pg8vleb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=f6d74d42-83ad-4f62-b66f-95220df7a691%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=3475cbcf-3485-4890-8793-94d52d50e342%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=5a436425-e7d3-47ae-a891-b82322bdb999%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=6ef2c76e-a156-4482-b188-21f228f029eb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://mdk06ma93j.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=741d2f76-2a64-4b48-8545-d32360915e33%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://rp2is7byfb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=5c5dd1c2-964a-4b58-b15b-5d24939cc01c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://mdk06ma93j.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=ce1209c3-8660-4137-8976-514219ce8e62%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://hu1pg8vleb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=1aa67118-c4bf-41c0-9937-049e7890afd7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://rp2is7byfb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=63a9b7f2-077b-4de6-be0f-2105ac31b06b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://hu1pg8vleb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=a25b3799-f861-4c9a-86aa-786172365893%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=d614222e-fb99-42f6-915d-5d8d71710d0c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://mdk06ma93j.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=579cc47e-37de-4248-b227-6af297898949%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://rp2is7byfb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=a86efcef-f0ef-419b-b2f7-173ad939227e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://hu1pg8vleb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=ccb37f60-8429-4b60-ba48-f304a01c5c91%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=9981ced0-ca7c-4ac8-91a6-6ba6ffe0164e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=533f8eec-08a1-4134-a761-7423c9c725d6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://mdk06ma93j.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=7beff4c4-e2b4-431c-8338-f742ab9c6438%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://rp2is7byfb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=9d7db29b-2e80-480f-af8c-da7761b48208%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://hu1pg8vleb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=afe25bdd-2bb6-434e-ac14-8677a0898f2b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=fc160524-ebf4-4753-b56a-04ef7683da95%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://mdk06ma93j.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=93c16523-44db-4372-9317-341834987aee%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://rp2is7byfb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=9774105b-564c-4dc7-9b6f-03f6bb8e4753%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://hu1pg8vleb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=a64e8aee-7ea7-4f33-b3a8-58b1612a20c5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=01b569c3-3ac4-4530-90a8-601941e69ea4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://mdk06ma93j.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=bad9da98-a0ff-4373-b3c3-fe6776df6014%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://rp2is7byfb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=a5be6ed9-690b-46e7-928e-d10f9eed8220%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://hu1pg8vleb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=9d75bb32-599a-445a-a406-16d92d1beb2c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=139945e3-928c-419c-89a0-aa5d1ed12e87%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://mdk06ma93j.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=bcfdac40-98a2-4851-ada9-2c5af981ff48%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://rp2is7byfb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=e352d48a-31df-4796-a850-fbbb73636a8d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://hu1pg8vleb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=9c5a9630-94aa-47c3-9bff-e1bcab19c783%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=4b82d5d4-5bde-4883-9354-97e1e047c418%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://mdk06ma93j.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=4eab28cd-aad1-4fd1-9f62-061a61387d19%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://rp2is7byfb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=7e5bf432-fa9e-4507-b1b9-ddbbfb7b86b4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://hu1pg8vleb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=e818c0fc-ba62-4a60-a2ac-467607badcb0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=f68b8a36-7502-415c-a632-80cc08b793c9%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: tests/integ/file_system_input_utils.py [hunk 1]
```diff
@@ -121,14 +121,15 @@ def _connect_ec2_instance(ec2_instance):
 
 def _upload_data_and_mount_fs(connected_instance, file_system_efs_id, file_system_fsx_id):
     connected_instance.put(FS_MOUNT_SCRIPT, ".")
-    connected_instance.run("mkdir temp_tf; mkdir temp_one_p")
+    connected_instance.run("mkdir temp_tf; mkdir temp_one_p", in_stream=False)
```
Comment (caxiaohu) [inline tests/integ/file_system_input_utils.py]: Nice catch!

File: src/sagemaker/inputs.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            "DataSource": {
+                "S3DataSource": {
+                    "S3DataDistributionType": distribution,
+                    "S3DataType": s3_data_type,
+                    "S3Uri": s3_data,
+                }
+            }
+        }
+
+        if compression is not None:
```
[the remark this answers is not in the document]
Comment (knakad) [reply src/sagemaker/inputs.py]: That's what I was thinking. Am I missing something that would make that incorrect?

File: tests/integ/file_system_input_utils.py [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+        error_code = e.response["Error"]["Code"]
+        message = e.response["Error"]["Message"]
+        if error_code == "NoSuchEntity" and ROLE_NAME in message:
+            return False
+        else:
+            raise
+    return True
+
+
+def retries(max_retry_count, exception_message_prefix):
```
[the remark this answers is not in the document]
Comment (knakad) [reply tests/integ/file_system_input_utils.py]: Completely agreed. Definitely no more changes in this PR, haha.

File: tests/integ/vpc_test_utils.py [hunk 4]
```diff
@@ -21,33 +21,107 @@
 LOCK_PATH = os.path.join(tempfile.gettempdir(), "sagemaker_test_vpc_lock")
 
 
-def _get_subnet_ids_by_name(ec2_client, name):
+def _subnet_ids_by_name(ec2_client, name):
```
[the remark this answers is not in the document]
Comment (knakad) [reply tests/integ/vpc_test_utils.py]: Do you have any readings on this that I could take advantage of? I wasn't able to find anything useful on the interwebs.

File: src/sagemaker/inputs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (laurenyu) [reply src/sagemaker/inputs.py]: boto3 maycomplain about parameters being set to `None`. The error will be something like "incorrect parameter type <NoneType>, should be one of: <str>"

File: tests/integ/vpc_test_utils.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (laurenyu) [reply tests/integ/vpc_test_utils.py]: I forget where I originally read/heard it from, though I see it mentioned a lot in StackOverflow answers. iirc the idea is that "get" is redundant to what's happening. @mvsusp, do you have any references handy on this one?

**Pull request merged**